salt/states/mac_defaults.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+'''
+Writing/reading defaults from an OS X minion
+=======================
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+
+# Import salt libs
+import salt.utils
+
+log = logging.getLogger(__name__)
+__virtualname__ = 'macdefaults'
+
+
+def __virtual__():
+    '''
+    Only work on Mac OS
+    '''
+    if salt.utils.is_darwin():
+        return __virtualname__
+    return False
+
+
+def write(name, domain, value, type='string', user=None):
+    '''
+    Write a default to the system
+
+    name
+        The key of the given domain to write to
+
+    domain
+        The name of the domain to write to
+
+    value
+        The value to write to the given key
+
+    type
+        The type of value to be written, vaid types are string, data, int[eger],
+        float, bool[ean], date, array, array-add, dict, dict-add
+
+    user
+        The user to write the defaults to
+
+
+    '''
+    ret = {'name': name,
+           'result': True,
+           'comment': '',
+           'changes': {}}
+
+    def safe_cast(val, to_type, default=None):
+        try:
+            return to_type(val)
+        except ValueError:
+            return default
+
+    current_value = __salt__['macdefaults.read'](domain, name, user)
+
+    if (type == 'bool' or type == 'boolean') and (
+                ((value is True or value is 'TRUE' or value is 'YES') and current_value == '1') or (
```

Comment:
There are a few issues here. First, we're overriding the use of the reserved keyword of `type` in Python, which should be avoided. Secondly, we should likely at least break this up into multiple conditionals for readability or consider the use of the `if x in [y,x]` syntax to make this easier to parse.